modules/videoio/misc/java/src/cpp/videoio_converters.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef VIDEOIO_CONVERTERS_HPP
+#define VIDEOIO_CONVERTERS_HPP
+
+#include <jni.h>
+#include "opencv_java.hpp"
+#include "opencv2/core.hpp"
+#include "opencv2/videoio/videoio.hpp"
+
+class JavaStreamReader : public cv::IStreamReader
+{
+public:
+    JavaStreamReader(JNIEnv* env, jobject obj);
```

Comment:
> The JNIEnv is used for thread-local storage. For this reason, you cannot share a JNIEnv between threads. If a piece of code has no other way to get its JNIEnv, you should share the JavaVM, and use GetEnv to discover the thread's JNIEnv. (Assuming it has one; see AttachCurrentThread below.)

Source: https://developer.android.com/training/articles/perf-jni?hl=en

Looks like the adapter should handle the threading model in smarter way.